slides/04-evaluating-models.qmd

```diff
@@ -22,69 +22,235 @@ knitr:
 #| file: setup.R
 ```
 
-## Metrics for model performance `r hexes("yardstick")`
+## Looking at predictions
 
 ```{r}
 #| echo: false
-library(tidymodels)
 library(countdown)
-data("tree_frogs", package = "stacks")
-tree_frogs <- tree_frogs %>%
-  mutate(t_o_d = factor(t_o_d),
-         age = age / 86400) %>% 
-  filter(!is.na(latency)) %>%
-  select(-c(clutch, hatched))
+library(tidymodels)
+library(modeldatatoo)
+taxi <- data_taxi() %>%
+  mutate(month = factor(month, levels = c("Jan", "Feb", "Mar", "Apr"))) %>% 
+  select(-c(id, duration, fare, tolls, extras, total_cost, payment_type)) %>%
+  drop_na()
 
 set.seed(123)
-frog_split <- initial_split(tree_frogs, prop = 0.8, strata = latency)
-frog_train <- training(frog_split)
-frog_test <- testing(frog_split)
-tree_spec <- decision_tree(cost_complexity = 0.001, mode = "regression")
-tree_wflow <- workflow(latency ~ ., tree_spec)
-tree_fit <- fit(tree_wflow, frog_train)
+taxi_split <- initial_split(taxi, prop = 0.8, strata = tip)
+taxi_train <- training(taxi_split)
+taxi_test <- testing(taxi_split)
+
+tree_spec <- decision_tree(cost_complexity = 0.00001, mode = "classification")
```

Comment:
```suggestion
tree_spec <- decision_tree(cost_complexity = 0.1, mode = "classification")
```

"What makes a model" currently uses a higher value so that we don't have to explain redundant splits with the lower ones.